src/react/connect.ts

```diff
@@ -0,0 +1,29 @@
+import { useStream, NOT_YET_EMITTED } from './useStream';
+import { Observable } from 'rxjs';
+import React, { ComponentType } from 'react';
+
+/**
+ * Higher order component for connecting a component to a stream
+ *
+ * The stream does not need to provide all the properties the component expects,
+ * any missing properties will be required by the component returned from this
+ * component.
+ *
+ * @param WrappedComponent Component that will recieve props from the stream
+ * @param stream$ Stream that will provide props to the component
+ * @see useStream
+ */
+export const connect = <Props, Observed extends Partial<Props>>(
+  Component: ComponentType<Props>,
+  stream$: Observable<Observed>
+) => (props: Omit<Props, keyof Observed>) => {
+  const value = useStream(stream$);
+
+  if (value === NOT_YET_EMITTED) return null;
+
+  // Typescript doesn't recognize this as Observed & Props for some reason
```

Comment:
If you use `Exclude` instead of `Omit`, does it work then?

Reply:
Check if this can be applied: https://github.com/ardoq/shared-ui/blob/eb13e2f926a7db3a80e71ad5f657a9bfce976414/src/utils.tsx#L31-L49

Reply:
`Omit` is a shorthand for the `Pick` and `Exclude` done on line 31 of your link.
The reason your example works, is that the added props are known at compile time.

See this SO answer: https://stackoverflow.com/a/56178198/1104307
And this issue in the TS bugtracker: https://github.com/microsoft/TypeScript/issues/36316